io/net/Network_config.py

```diff
@@ -0,0 +1,127 @@
+        self.log.info(driver)
+        cmd = "ethtool -i %s | grep bus-info | awk '{print $2}'" % self.iface
+        businfo = process.system_output(cmd, shell=True).strip()
+        self.log.info(businfo)
+        cmd = "lspci -v"
+        bus_info = process.system_output(cmd, shell=True).strip()
+        bus_info = bus_info.split('\n\n')
+        self.log.info("Performing driver match check using lspci and ethtool")
+        self.log.info("-----------------------------------------------------")
+        for value in bus_info:
+            if value.startswith(businfo):
+                self.log.info("details are ---------> %s" % value)
+                tmp = value.split('\n')
+                for val in tmp:
+                    if 'Kernel driver in use' in val:
+                        cmd, driverinfo = val.split(': ')
+                        self.log.info(driverinfo)
+                        if driver != driverinfo:
+                            self.fail("mismatch in driver information")
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/mtu" %\
+              (driver, driver, businfo, self.iface)
+        mtu = process.system_output(cmd, shell=True).strip()
+        self.log.info("mtu value is %s" % mtu)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/operstate" %\
+              (driver, driver, businfo, self.iface)
+        operstate = process.system_output(cmd, shell=True).strip()
+        self.log.info("operstate is %s" % operstate)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/duplex" %\
+              (driver, driver, businfo, self.iface)
+        duplex = process.system_output(cmd, shell=True).strip()
+        self.log.info("transmission mode is %s" % duplex)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/address" %\
+              (driver, driver, businfo, self.iface)
+        address = process.system_output(cmd, shell=True).strip()
+        self.log.info("mac address is %s" % address)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/speed" %\
+              (driver, driver, businfo, self.iface)
+        speed = process.system_output(cmd, shell=True).strip()
+        self.log.info("speed is %s" % speed)
+        cmd = "ethtool %s | grep Speed | awk '{print $2}'" % self.iface
+        eth_speed = process.system_output(cmd, shell=True).strip()
+        eth_speed = eth_speed.strip('Mb/s')
+        self.log.info("Performing Ethtool and interface checks for interface")
+        self.log.info("-----------------------------------------------------")
+        if speed != eth_speed:
+            self.fail("mis match in speed")
+        hw_addr = netifaces.ifaddresses(self.iface)[netifaces.AF_LINK]
+        hw_addr = hw_addr[0]['addr']
+        if hw_addr != address:
+            self.fail("mismatch in hardware address")
+        cmd = "ifconfig %s" % self.iface
+        mtuval = process.system_output(cmd, shell=True).strip()
```

Comment:
can this be 

`mtuval = process.system_output(cmd, shell=True).strip().split('\n')` to eliminate extra variable